ethcore/src/client.rs

```diff
@@ -404,8 +404,12 @@ impl<V> Client<V> where V: Verifier {
 
 		{
 			let mut block_queue = self.block_queue.write().unwrap();
-			block_queue.mark_as_bad(&bad_blocks);
-			block_queue.mark_as_good(&good_blocks);
+			if !bad_blocks.is_empty() {
```

Comment:
I would rather leave comment Inside to explain why it's better not to remove it (or maybe it's possible to write test that empty list should rearrange?)

But LGTM anyway.